Eve 5.8.2 doesn't reboot after onboarding

# Description

Eve doesn't reboot after onboarding. Using Adam and Eden 

# Steps to Reproduce

for Eden: 

make build
./eden config add default
./eden config set default --key=eve.tag --value=5.8.2
./eden test ./tests/workflow

# Actual Results

Eden onboard (3/11)
time="2020-08-07T09:33:50Z" level=info msg="testData directory: testdata"
edenProg: /var/lib/jenkins/workspace/eve-docker-image/dist/bin/eden timewait: 0s
testProg: /var/lib/jenkins/workspace/eve-docker-image/dist/bin/eden.reboot.test timewait: 0s
--- FAIL: TestEdenScripts (0.00s)
    --- FAIL: TestEdenScripts/eden_onboard (1945.11s)
        testscript.go:392: 
            # eden start
            # Onboarding. (120.077s)
            # Waiting of onboard reboot. (1825.034s)
            > test eden.reboot.test -test.v -timewait 1800 -reboot=0 -count=1 &
            testProg: /var/lib/jenkins/workspace/eve-docker-image/dist/bin/eden.reboot.test
            > wait
            [background] eden.reboot.test -test.v -timewait 1800 -reboot=0 -count=1 &: exit status 1
            [stdout]
            Reboot Test
            === RUN   TestReboot
            --- FAIL: TestReboot (1825.00s)
                reboot_test.go:123: Wait for state of node1
                reboot_test.go:125: timewait:  1800
                reboot_test.go:126: reboot:  false
                reboot_test.go:127: count:  1
                testContext.go:295: done waiting for state
                reboot_test.go:133: lastRebootTime: 2020-08-07T09:34:59.854227388Z
                reboot_test.go:146: Wait for reboot of node1
                testContext.go:226: WaitForProc terminated by timeout 30m0s
                reboot_test.go:153: Reboot number: 1
                reboot_test.go:156: Number of reboots: 1
            FAIL
            [stderr]
            time="2020-08-07T09:35:50Z" level=info msg="config updated /var/lib/jenkins/.eden/devUUID-95ff4233-6a25-4821-9cf8-783ebc805164.json"
            time="2020-08-07T09:35:50Z" level=info msg="state updated /var/lib/jenkins/.eden/state-2b00c73f-adac-435d-a8bb-05a54db02d0c.yml"
            FAIL: testdata/eden_onboard.txt:9: unexpected command failure


# Expected Results

Test passed


# Additional Information

Eve onboards, but doesn't reboot. Likely to be valid for Adam and v1 API   

<!-- e.g. issue happens only occasionally -->

--- a/pkg/pillar/pubsub/socketdriver/driver.go
+++ b/pkg/pillar/pubsub/socketdriver/driver.go
@@ -9,6 +9,7 @@ import (
 	"net"
 	"os"
 	"path"
+	"strings"
 
 	"github.com/lf-edge/eve/pkg/pillar/pubsub"
 	log "github.com/sirupsen/logrus"
@@ -162,14 +163,22 @@ func (s *SocketDriver) Subscriber(global bool, name, topic string, persistent bo
 	// Special case for files in /var/tmp/zededa/ and also
 	// for zedclient going away yet metrics being read after it
 	// is gone.
-	agentName := name
+	var agentName string
+	names := strings.Split(name, "/")
+	if len(names) > 0 {
+		agentName = names[0]
+	}
 
 	if global {
 		subFromDir = true
 		dirName = s.fixedDirName(name)
 	} else if agentName == "zedclient" {
 		subFromDir = true
-		dirName = s.pubDirName(name)
+		if persistent {
+			dirName = s.persistentDirName(name)
+		} else {
+			dirName = s.pubDirName(name)
+		}
 	} else if persistent {
 		// We do the initial Load from the directory if it
 		// exists, but subsequent updates come over IPC

